static/examples/batch/batch-decomposition-threads.scd

```diff
@@ -0,0 +1,74 @@
+(
+s.reboot;
+s.waitForBoot({
+    // Store the location of the FluCoMa audio files in the audioFiles variable
+    var audioFiles = PathName(File.realpath(FluidDataSet.class.filenameSymbol).dirname.withTrailingSlash ++ "../AudioFiles/");
+    // Create a dictionary that will contain the resulting buffer, associated to a string which represents the original source file
+    var results = Dictionary.new;
+
+    // Create a condition that hangs while processing is still happening and unhangs when its complete.
+    var processingCompleted = Condition();
+
+    // Maintain a counter for the number of files processed.
+    // This will be use to determine when all of the files are finished working.
+    var counter = 0;
+
+    // Remove any files from the audioFiles collection so that only those with a .wav extension remain
+    var wav = audioFiles.files.removeAllSuchThat({ |file| file.extension == "wav" });
+
+    // In the language iterate over all the files in the example audio files directory
+    wav.do({ |file|
+        fork ({
```

Comment:
This will potentially start a lot of worker threads on the server, so it could actually be quite slow. There isn't a nice solution to this yet. The safest / most paternalistic would be to have a FluCoMa-wide thread pool to cap the number of OS threads in flight for async jobs, but that's decidedly non-trivial. Next best is to cap from the language, but that's not hugely simple either (I guess using a `Semaphore` might do it)

Reply:
Yeah this is only pleasant with these small jobs...

I'll looking into the `Semaphore` and update accordingly.